src/apify/storage_clients/_apify/_key_value_store_client.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import asyncio
+from logging import getLogger
+from typing import TYPE_CHECKING, Any
+
+from typing_extensions import override
+from yarl import URL
+
+from apify_client import ApifyClientAsync
+from crawlee.storage_clients._base import KeyValueStoreClient
+from crawlee.storage_clients.models import KeyValueStoreMetadata, KeyValueStoreRecord, KeyValueStoreRecordMetadata
+
+from ._models import KeyValueStoreListKeysPage
+from apify._crypto import create_hmac_signature
+
+if TYPE_CHECKING:
+    from collections.abc import AsyncIterator
+
+    from apify_client.clients import KeyValueStoreClientAsync
+
+    from apify import Configuration
+
+logger = getLogger(__name__)
+
+
+class ApifyKeyValueStoreClient(KeyValueStoreClient):
+    """An Apify platform implementation of the key-value store client."""
+
+    def __init__(
+        self,
+        *,
+        api_client: KeyValueStoreClientAsync,
+        api_public_base_url: str,
+        lock: asyncio.Lock,
+    ) -> None:
+        """Initialize a new instance.
+
+        Preferably use the `ApifyKeyValueStoreClient.open` class method to create a new instance.
+        """
+        self._api_client = api_client
+        """The Apify KVS client for API operations."""
+
+        self._api_public_base_url = api_public_base_url
+        """The public base URL for accessing the key-value store records."""
+
+        self._lock = lock
+        """A lock to ensure that only one operation is performed at a time."""
+
+    @override
+    async def get_metadata(self) -> KeyValueStoreMetadata:
+        metadata = await self._api_client.get()
+        return KeyValueStoreMetadata.model_validate(metadata)
```

Comment:
Could this use an extended model that would include `urlSigningSecretKey`?

Reply:
sure